repository_service_tuf/helpers/tuf.py

```diff
@@ -181,18 +174,29 @@ def _update_snapshot(
                 f"signing threshold '{threshold}'"
             )
 
-        root.roles[role_name] = Role([], threshold)
-        for signer in signers:
-            root.add_key(
-                Key.from_securesystemslib_key(signer.key_dict), role_name
-            )
+        roles[role_name] = Role([], threshold)
+        add_key_args: list[tuple[Key, str]] = [
```

Comment:
Can we postpone merging, I am not very confident about this line. This is in a for loop for every role name in `TOP_LEVEL_ROLE_NAMES`, but we are re-writing to the `add_key_args` variable before passing them to `root.add_key`

Reply:
Functional tests are good.
The important part is the signing and bootstrap, which comes from the CLI. They are OK. The overall tests are also passing.
```shell
❯ pytest -vv ../repository-service-tuf-cli/tests/unit/cli/admin/test_ceremony.py::TestCeremonyGroupCLI::test_ceremony_start_default_values
========================================================================== test session starts ===========================================================================
platform darwin -- Python 3.10.0, pytest-7.2.1, pluggy-1.0.0 -- /Users/kdearaujo/.local/share/virtualenvs/repository-service-tuf-E73fsQ7i/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.10.0', 'Platform': 'macOS-12.6.1-x86_64-i386-64bit', 'Packages': {'pytest': '7.2.1', 'pluggy': '1.0.0'}, 'Plugins': {'html': '3.2.0', 'bdd-html': '0.1.14a0', 'metadata': '2.0.4', 'bdd': '6.1.1'}}
rootdir: /Users/kdearaujo
plugins: html-3.2.0, bdd-html-0.1.14a0, metadata-2.0.4, bdd-6.1.1
collected 1 item                                                                                                                                                         

../repository-service-tuf-cli/tests/unit/cli/admin/test_ceremony.py::TestCeremonyGroupCLI::test_ceremony_start_default_values PASSED                               [100%]

=========================================================================== 1 passed in 0.79s ============================================================================
❯ time rstuf -c rstuf.ini admin ceremony -b -u -s -f payload.json
Starting online bootstrap
Bootstrap status: ACCEPTED (847f723b10944f19bde3457cc82c5785)
Bootstrap status: STARTED
Bootstrap status: SUCCESS
Bootstrap finished.

Ceremony done. 🔐 🎉
rstuf -c rstuf.ini admin ceremony -b -u -s -f payload.json  0.38s user 0.14s system 12% cpu 4.235 total
❯ make functional-tests
===================================================================== 68 passed in 368.66s (0:06:08) =====================================================================
```

Reply:
Alright then, thanks ☺️